src/System.IO/src/System/IO/StreamReader.cs

```diff
@@ -480,6 +521,14 @@ private void DetectEncoding()
                 CompressBuffer(3);
                 changedEncoding = true;
             }
+            else if (_byteLen >= 4 && _byteBuffer[0] == 0 && _byteBuffer[1] == 0 &&
+                _byteBuffer[2] == 0xFE && _byteBuffer[3] == 0xFF)
+            {
+                // Big Endian UTF32
+                _encoding = new UTF32Encoding(true, true);
```

Comment:
Thanks for the quick feedback on the update! I pushed another with the parameter names. The GetEncoding use looks good, but I'd rather we do that in a different PR and keep this one  as minimum a set of porting changes as possible.